tests/test_columns.py

```diff
@@ -222,3 +223,21 @@ def _get_view_data(self, view):
         vc = QueryContext("user_table")
         vc.append_column(view)
         return vc.resolve(self.session.connection())
+
+    def test_distinct(self):
+        vc = QueryContext(
+            "user_table",
+            distinct=['user', 'year'],
+            order_by=[OrderBy('user'), OrderBy('year'), OrderBy('date', is_ascending=False)],
+            group_by=['user', 'date']
+        )
+        vc.append_column(SimpleColumn('user'))
+        vc.append_column(YearColumn('date', alias='year'))
+        vc.append_column(SimpleColumn('indicator_a'))
+        result = vc.resolve(self.session.connection())
+        self.assertEquals(result, {
+            ('user1', datetime.date(2013, 2, 1)): {'user': 'user1', 'year': 2013.0, 'indicator_a': 3,
```

Comment:
It's weird that a year value has a `.0`.

Reply:
I think some parsing is done later but ya its weird.